packages/web-app-files/src/helpers/resources.ts

```diff
@@ -49,6 +49,16 @@ export function buildResource(resource): Resource {
   }
 
   const id = resource.fileInfo[DavProperty.FileId]
+  const idComponent = resource.name.split('/')[2]
+  let webDavPath
+  if (idComponent?.includes('!')) {
+    // idComponent includes the opaqueID which represents the resource name.
+    // For the webdav path, we need to replace the opaqueID with the actual resource name.
+    const withoutOpaqueId = resource.name.split('!')[0]
+    webDavPath = `${withoutOpaqueId}/${name}`
+  } else {
+    webDavPath = resource.name
+  }
```

Comment:
I don't know here... maybe there is a more elegant way to solve this issue? With alias links on single resources, we don't have the actual resource name in the path, but the opaque id. Actions like `rename` or editing via the text editor require the actual name in the `webDavPath` though.

Reply:
the sdk has a function to retrieve a webdav path by file id via `meta` request. maybe you need to do that...

Reply:
But that would be one additional request per resource, right...?

Reply:
ah, per resource... that's not an option.

Reply:
I adjusted the condition a bit so it only affects alias links on single resources. This is a very special case as described above, I don't see another way right now besides moving this logic to all places where needed (rename, text editor app, ...).

Reply:
Uhm, what logic would you want to add to text editor? What makes the text editor special?